test/multiple-user-principal-types.test.js

```diff
@@ -54,338 +69,295 @@ describe('Multiple users with custom principalType:', function() {
     }
+    expect(accessToken.id, 'token id').to.have.lengthOf(64);
     expect(accessToken).to.have.property('userId', user.id);
     expect(accessToken).to.have.property('principalType', user.constructor.definition.name);
-    assert(accessToken.id);
-    assert.equal(accessToken.id.length, 64);
   }
 
   describe('User.logout', function() {
     it('should logout a user from user model 1 without logging out user from model 2',
-    function(done) {
-      var userFromOneModelTokens, userFromAnotherModelTokens;
-      Promise.all([
+    function() {
+      var tokenOfOneUser, tokenOfAnotherUser;
+      return Promise.all([
         OneUser.login(commonCredentials),
         AnotherUser.login(commonCredentials),
       ])
-      .spread(function(at1, at2) {
-        userFromOneModelTokens = at1;
-        userFromAnotherModelTokens = at2;
-        return OneUser.logout(userFromOneModelTokens.id);
+      .spread(function(t1, t2) {
+        tokenOfOneUser = t1;
+        tokenOfAnotherUser = t2;
+        return OneUser.logout(tokenOfOneUser.id);
       })
       .then(function() {
         return Promise.all([
-          AccessToken.findById(userFromOneModelTokens.id),
-          AccessToken.findById(userFromAnotherModelTokens.id),
+          AccessToken.findById(tokenOfOneUser.id),
+          AccessToken.findById(tokenOfAnotherUser.id),
         ]);
       })
-      .spread(function(userFromOneModelTokens, userFromAnotherModelTokens) {
-        assert(!userFromOneModelTokens, 'accessToken should not exist after logging out user1');
-        assert(userFromAnotherModelTokens, 'accessToken should exist after logging out user1');
-        done();
-      })
-      .catch(done);
+      .spread(function(tokenOfOneUser, tokenOfAnotherUser) {
+        expect(tokenOfOneUser).to.not.exist();
+        expect(tokenOfAnotherUser).to.exist();
+      });
     });
   });
 
-  describe('Password Reset: ', function() {
+  describe('Password Reset', function() {
     describe('User.resetPassword(options)', function() {
       var options = {
         email: 'foo@bar.com',
         redirect: 'http://foobar.com/reset-password',
       };
 
-      it('should create a temp accessToken to allow a user to change password', function(done) {
-        OneUser.resetPassword({email: options.email})
-        .catch(function(err) {
-          assert(!err);
-        });
-
-        OneUser.once('resetPasswordRequest', function(info) {
-          assertGoodToken(info.accessToken, userFromOneModel);
-          info.accessToken.user.getAsync()
-          .then(function(user) {
-            assert.equal(user.id, userFromOneModel.id);
-            assert.equal(user.email, userFromOneModel.email);
-            done();
+      it('should create a temp accessToken to allow a user to change password', function() {
+        return Promise.all([
+          OneUser.resetPassword({email: options.email}),
+          waitForEvent(OneUser, 'resetPasswordRequest')
```

Comment:
Hmm, this makes it rather difficult to see what are the arguments of `Promise.all` and what is just a part of one of the async branches.

How about the following?

```js
return Promise.all([
  OneUser.resetPassword(...),
  waitForResetRequestAndVerify()
]);

function waitForResetRequestAndVerify() {  
  return waitForEvent(...)
   // ...
}
```